ExampleMod/Content/Projectiles/ExampleSineWaveMotionProjectile.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using System;
+using Terraria;
+using Terraria.DataStructures;
+using Terraria.GameContent;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Content.Projectiles
+{
+	// This file showcases a projectile moving in a sine wave
+	// Can be tested with ExampleMagicRod
+	public class ExampleSineWaveMotionProjectile : ModProjectile
+	{
+		// This field is used in this projectie's custom AI
+		public Vector2 initialCenter;
+
+		// This field is used as a counter for the wave motion
+		public int sineTimer;
+
+		public int waveDirection = 1;
+
+		public Color drawColor = Color.Red;
+
+		public override void SetStaticDefaults() {
+			DisplayName.SetDefault("Example Sine Wave Motion Projectile"); // Name of the projectile. It can be appear in chat
+
+			Main.projFrames[Type] = 3; // This projectile has 3 frames in its spritesheet
+		}
+
+		// Setting the default parameters of the projectile
+		// You can check most of Fields and Properties here https://github.com/tModLoader/tModLoader/wiki/Projectile-Class-Documentation
+		public override void SetDefaults() {
+			Projectile.width = 8; // The width of projectile hitbox
+			Projectile.height = 8; // The height of projectile hitbox
+
+			Projectile.aiStyle = 0; // The ai style of the projectile (0 means custom AI). For more please reference the source code of Terraria
+			Projectile.DamageType = DamageClass.Magic; // What type of damage does this projectile affect?
+			Projectile.friendly = true; // Can the projectile deal damage to enemies?
+			Projectile.hostile = false; // Can the projectile deal damage to the player?
+			Projectile.ignoreWater = true; // Does the projectile's speed be influenced by water?
+			Projectile.light = 1f; // How much light emit around the projectile
+			Projectile.tileCollide = true; // Can the projectile collide with tiles?
+			Projectile.timeLeft = 600; // The live time for the projectile (60 = 1 second, so 600 is 10 seconds)
+		}
+
+		// This projectile updates its position manually
+		public override bool ShouldUpdatePosition() {
+			return false;
+		}
+
+		public override void OnSpawn(IEntitySource source) {
+			initialCenter = Projectile.Center;
+		}
+
+		public override Color? GetAlpha(Color lightColor) {
+			return drawColor * 0.64f;
+		}
+
+		public override void AI() {
+			// Ensure that the "waveDirection = 1" projectile starts above the "waveDirection = -1" projectile not matter which direction they were fired from
+			int trueDirection = Projectile.direction * waveDirection;
```

Comment:
Why use `Projectile.direction` here? I don't see where it's set.

Reply:
Isn't it set automatically?

Reply:
I double checked, and it is.

Reply:
nevermind on the previous two comments.  `Projectile.direction` is set later in AI